### DISCUSSION TITLE
139

### DISCUSSION THREAD

**Discussion by jojoldu** [jojoldu/blog-comments#599] [Comments]
http://jojoldu.tistory.com/139

**Comment by jojoldu** [comment]
작성자: duckgu

slack 까지 연동하는 건 생각을 못했었는데
좋은 정보 감사합니다.

**Comment by jojoldu** [comment]
작성자: 캥캥

좋은 정보 감사합니다.
오늘 한번 따라해보아야 겠네요.

ps.오타?
1) 링크를 통해 젠킨스를 설치하셨다면 도커의 Kitematic을 실행합니다.  -> 링크를 통해 도커를  이 맞는것 같은데.. 
2) 입려이 완료되면 Credentials 선택창에 방금 등록한 계정정보가 출력되니, 이를 선택합니다.  ->입력
3) step3의 Team Domain과 Integration Token을 복사해놓습니다. 부분의 step3 그림에  Integration Token 이 보이는것 같네요.

**Comment by jojoldu** [comment]
작성자: seungdols

좋은 블로깅 감사합니다!!^^ 이제는 도커까지 발을 넓히시는 군요!!

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

ㅋㅋ 아닙니다 아직 많이 부족합니다 ㅠ
인프라 지식이 얕아서 도커로 조금씩 연습해보려구요! ㅎㅎ
승돌님께서 올려주시는 블로그글 매번 잘보고 있습니다^^
항상 감사합니다!

**Comment by jojoldu** [comment]
작성자: 박수영

동욱님! 정말 좋은글 잘 읽었습니다! 잘 되서 괜히 기뿌네요 ㅎㅎ  친절한 설명 항상 너무 고맙습니다 :)

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

ㅎㅎ 감사합니다 수영님! 수영님도 좋은 회사 입사하셔서 정말 축하드립니다! 
slipp 스터디 활동도 하시고 정말 보기 좋습니다!
매번 감사드립니다^^

**Comment by jojoldu** [comment]
작성자: guppy

와 역시이십니다..

**Comment by jojoldu** [comment]
작성자: 제리루즈

안녕하세요 창천항로님
매번 좋은글 감사합니다. 쉽게 설명해주셔서 잘 따라하구 있습니다.
실습중에 한가지 질문이 있습니다.
젠킨스 설정후에 빌드 시 아래와 같은 에러로그가 뜨는데..
gradlew를 찾지 못하는거 같습니다.

+ ./gradlew clean print
/tmp/jenkins2690807495926315910.sh: 2: /tmp/jenkins2690807495926315910.sh: ./gradlew: not found
Build step 'Execute shell' marked build as failure
Finished: FAILURE

프로젝트는 sts에서 스프링부트로 생성해서 gradlew 파일이 있고 github에서 있는거 확인했는데
왜 찾지 못하는걸까요!?
gradle도 그렇고 jenkins도 처음 써보는거라 모르는게 많네요ㅠ_ㅠ
괜찮으시면 조언 부탁드립니다!

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

안녕하세요!
혹시 github에 gradlew 파일이 올라가있는지 확인해보시겠어요?
해당 젠킨스는 결국 Github에 올라간 파일들을 다 받아와 빌드를 수행합니다.
확인 부탁드리겠습니다^^;

**Comment by jojoldu** [comment]
작성자: 제리루즈

네 github 저장소에 보면 gradlew 파일이 존재합니다.
프로젝트 폴더를 열어보면
bin
gradle/wrapper
src
.gitignore
buil.gradle
gradlew
gradlew.bat

위와 같이 폴더와 파일들이 존재하는데 gradlew 파일도 같이 있거든요..ㅠㅠ
혹시 로컬에 gradle을 따로 설치해서 환경변수를 잡아주고 하는게 필요한가요?
그냥 sts설치하고 gradle support 플러그인만 깔고 진행한거거든요ㅎㅎ

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

아! 그러시군요
음 그럼 git에서 파일을 잘 받았는지를 확인해봐야겠네요
젠킨스 도커 컨테이너에 접속하셔서
/var/jenkins_home/workspace/springboot 
위치에 gradlew가 있는지 확인해보시겟어요?

**Comment by jojoldu** [comment]
작성자: 제리루즈

말씀해주신 경로를 보니 springboot-jenkins-docker 폴더가 있네요!?(프로젝트 폴더인듯하네요)

/var/jenkins_home/workspace/springboot/springboot-jenkins-docker/gradlew

파일이 있네요ㅎㅎ;
중간에 프로젝트명으로된 폴더가 하나 더 끼어있어서 파일을 못찾은건가보네요
고대로 따라했는데 왜 끼어있을까요;;

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

ㅎㅎ 아마 git이 springboot 프로젝트 단위로 묶여서 그러신듯 합니다.
gradlew위치를 ./springboot-jenkins-docker/gradlew

로 변경해주시면 될것 같습니다^^;

**Comment by jojoldu** [comment]
작성자: 제리루즈

아하 빠르고 상세한답변 감사합니당ㅎㅎ

굿밤되세요!!

**Comment by jojoldu** [comment]
작성자: 달란


먼저 좋은 글 감사합니다.
처음으로 CI 구축하는데 정말 큰 도움이 되었습니다.
다름이 아니라 github 설정에서 hook url 입력시 ngrok로 포워딩된 호스트를 입력해 주는 방법 말고는 없는 건가요?
컴터 부팅 후 다시 시도할시에 계속 url값을 변경해 주어야 하는 점 떄문에 질문 드립니다ㅜㅜ

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

안녕하세요!
방문해주셔서 감사합니다^^
말씀하신대로 ngrok 없이 진행하려면 github이 인식할 수 있도록 외부 도메인이 필요합니다 ㅠ
결국 젠킨스를 외부 호스팅 서버에 설치하거나 (카페24, 가비아 등) 집에 자체 서버를 구축해서 IP를 도메인에 연결시키는 등의 방법밖에 없습니다.
ngrok을 사용한것은 이렇게 하기가 싫어 사용한 방법입니다^^;;

도움이 안된것 같아 죄송합니다 ㅠ

**Comment by jojoldu** [comment]
작성자: ごるげつ

다른 게시물도 잘보고 있습니다.
maven 으로 빌드 시 에는 버전설정이 필요하군요.
프로젝트 시작하기 앞서 많은 도움이 되었습니다.

**Comment by jojoldu** [comment]
작성자: 봉구리

안녕하세요 혹시 docker로 띄운 jenkins가 
서버가 뻗는다거나 문제가 생길시 새로 증설하려하면
기존의 데이터를 유지하고 띄우려면 어떻게 해야 하나요?

**Comment by jojoldu** [comment]
작성자: King Tora

docker 이미지에서 새로 컨테이너를 생성하시면 됩니다!
실제로 운영을 하시게 되면 컨테이너의 변경점을 (제대로 다 되면) 이미지에 적용시켜서 이미지도 업데이트(업데이트시킨 이미지를 새로 생성합니다) 시킵니다. 그럼 추후에 컨테이너에서 작업하다가 문제가 생기면 컨테이너를 날리고 가장 최근에 업데이트한 이미지에서 새로 발급받으면 됩니다.

**Comment by luckymagic7** [comment]
안녕하세요. 로컬 서버에 젠킨스를 설치해서 실행하면 webhook 기능 줄 때도 github 사이트에 그냥
`http://localhost:포트번호/github-webhook/`
이런식으로 설정하면 되는 거 아닌가요? `git push`할 때 빌드 되지 않아서 여쭤봅니다..

**Comment by luckymagic7** [comment]
`ip:포트번호` 식으로 주니 안되고 ngrok을 실행해야지만 되네요.. ㅠ 매번 개인 서버를 재부팅 할 때마다, url이 달라지는군요 ㅠㅠ

**Comment by sarguments** [comment]
안녕하세요. "링크를 통해 젠킨스를 설치하셨다면 도커의 Kitematic을 실행합니다." 부분에서 젠킨스가 도커로 변경되어야 할 것 같습니다.

**Comment by jojoldu** [comment]
@sarguments 오타 확인 감사합니다 :)
수정했습니다 :)

**Comment by yangseungin** [comment]
평소 글만 열심히보다 처음으로 댓글달아보네요ㅎㅎ
CI 구축하는데 큰 도움이 되었습니다. 예전글이라 지금이랑 꽤나 바뀐 부분들이 있었지만 참고하였습니다.
좋은 글 앞으로도 많이써주세요 :)
